src/include/mpir_request.h

```diff
@@ -310,6 +318,9 @@ static inline int MPIR_Request_is_active(MPIR_Request * req_ptr)
             case MPIR_REQUEST_KIND__PREQUEST_SEND:
             case MPIR_REQUEST_KIND__PREQUEST_RECV:
                 return (req_ptr)->u.persist.real_request != NULL;
+            case MPIR_REQUEST_KIND__PART_SEND:
+            case MPIR_REQUEST_KIND__PART_RECV:
+                return MPID_Part_request_is_active(req_ptr);
```

Comment:
I don't quite understand why this (`MPID_Part_request_is_active`) need be device function call. Can't we simply use a atomic flag?

Could you clarify when is the request considered active? Is it between `MPI_Start` and `MPI_Test`? If it is, then it is entirely a MPIR-layer flag.

Reply:
Yes, it is active between `start`-`test|wait`. I think how to implement the "active" status is device-internal, e.g., for persistent req, it is based on whether there is an active req. So I put it inside device.

Having said that, I don't see any stronger reason to hide it inside device. I am OK to move it to `MPIR`, if it simplifies the code.

Reply:
moved `active_flag` to MPIR_Request; thus deleted `MPID_Part_request_is_active` and `MPID_Part_request_complete_hook`